vos/vos/tests/test_md5_cache.py

```diff
@@ -49,20 +52,18 @@ def test_sqlite3(self, mock_sqlite3):
            'SELECT md5, st_size, st_mtime FROM md5_cache WHERE fname = ? ', ('somefile',))
 
 
-    @patch('__builtin__.open')
-    def test_computeMD5(self, mock_file):
-        md5_cache = MD5_Cache()
+    def test_computeMD5(self):
         file_mock = MagicMock()
-        buffer = ['abcd', 'efgh', '']
+        buffer = [b'abcd', b'efgh', b'']
         file_mock.read.side_effect = buffer
-        mock_file.return_value = file_mock
-        
+
         expect_md5 = hashlib.md5()
         for b in buffer:
             expect_md5.update(b)
         
         md5_cache = MD5_Cache()
-        self.assertEquals(expect_md5.hexdigest(), md5_cache.computeMD5('fakefile', 4))
+        with patch('six.moves.builtins.open', mock_open(read_data=b''.join(buffer))):
```

Comment:
This is a way to join binaries into a binary string. Without b, it tries to join them as strings.